lib/licensed/sources/npm.rb

```diff
@@ -135,6 +137,18 @@ def yarn_lock_present
       def include_non_production?
         config.dig("npm", "production_only") == false
       end
+
+      def missing_peer?(parent, dependency, name)
+        dependency["peerMissing"] || (dependency["missing"] && peer_dependency(parent, name))
+      end
+
+      def peer_dependency(parent, name)
+        parent.dig("peerDependencies", name)
```

Comment:
it's possible for parent to be nil if this gets called from the `recursive_dependencies` call on the root dependencies at line 54. It's probably a good idea to update that call to include `package_metadata` as the parent, but for extra safety this should use the nil safe operator

```suggestion
        parent&.dig("peerDependencies", name)
```